libs/blocks/quiz/quiz.js

```diff
@@ -0,0 +1,338 @@
+import {
+  initConfigPathGlob, handleResultFlow, handleNext, transformToFlowData, getQuizData,
+  getAnalyticsDataForBtn, getUrlParams,
+} from './utils.js';
+import StepIndicator from './stepIndicator.js';
+
+export async function loadFragments(fragmentURL) {
+  const quizSections = document.querySelector('.quiz-footer');
+  const a = createTag('a', { href: fragmentURL });
+  quizSections.append(a);
+  const { default: createFragment } = await import('../fragment/fragment.js');
+  await createFragment(a);
+}
+
+const App = ({
+  initialIsDataLoaded = false,
+  preQuestions = {}, initialStrings = {},
+}) => {
+  const [questionData, setQuestionData] = useState(preQuestions.questionData || {});
+  const [stringData, setStringData] = useState(initialStrings || {});
+  const [isDataLoaded, setDataLoaded] = useState(initialIsDataLoaded);
+  const [selectedQuestion, setSelectedQuestion] = useState(preQuestions || null);
+  const [questionList, setQuestionList] = useState(preQuestions.questionList || {});
+  const [stringQList, setStringQList] = useState(preQuestions.stringQList || {});
+  const [userSelection, updateUserSelection] = useState([]);
+  const [userFlow, setUserFlow] = useState([]);
+  const [selectedCards, setSelectedCards] = useState({});
+  const [countSelectedCards, setCountOfSelectedCards] = useState(0);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [totalSteps, setTotalSteps] = useState(3);
+  const [prevStepIndicator, setPrevStepIndicator] = useState([]);
+  const [nextQuizViewsExist, setNextQuizViewsExist] = useState(true);
+  const [btnAnalytics, setBtnAnalytics] = useState(null);
+  const [btnClicked, setBtnClicked] = useState(false);
+  const initialUrlParams = getUrlParams();
+  const [urlParam, setUrlParam] = useState(initialUrlParams);
+  const VALID_QUESTIONS = useMemo(() => [], []);
+  const KNOWN_PARAMS = useMemo(() => ['martech', 'milolibs', 'quiz-data'], []);
+
+  useEffect(() => {
+    (async () => {
+      const [questions, dataStrings] = await getQuizData();
+      const qMap = {};
+      questions.questions.data.forEach((question) => {
+        qMap[question.questions] = question;
+        VALID_QUESTIONS.push(question.questions);
+      });
+
+      const strMap = {};
+      dataStrings.questions.data.forEach((question) => {
+        strMap[question.q] = question;
+      });
+
+      setUserFlow([questions.questions.data[0].questions]);
+
+      setStringData(dataStrings);
+      setQuestionData(questions);
+      setStringQList(strMap);
+      setQuestionList(qMap);
+
+      // wait for data to load
+      setDataLoaded(true);
+
+      // add quiz class to page
+      document.body.classList.add('quiz-page');
+    })();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setQuestionData, setStringData, setStringQList, setQuestionList]);
+
+  useEffect(() => {
+    function handlePopState() {
+      const currentURL = window.location.href;
+      window.location.href = currentURL;
```

Comment:
nit - also the function name doesn't seem right - should it just be something like `reloadPage`?
```suggestion
      window.location.reload()
```